ddl/ddl_db_test.go

```diff
@@ -2498,41 +2500,126 @@ func (s *testDBSuite) TestAlterTableAddPartition(c *C) {
 	partition p2 values less than maxvalue
 	);`)
 
-	sql3 := `alter table t2 add partition (
+	sql3 := `alter table table2 add partition (
 		partition p3 values less than (2010)
 	);`
-	s.testErrorCode(c, sql3, mysql.ErrPartitionMaxvalue)
+	s.testErrorCode(c, sql3, tmysql.ErrPartitionMaxvalue)
 
-	s.tk.MustExec("drop table if t3")
-	s.tk.MustExec(`create table t3 (
+	s.tk.MustExec("drop table if exists table3;")
+	s.tk.MustExec(`create table table3 (
 	id int not null,
 	hired date not null
 	)
 	partition by range( year(hired) ) (
 	partition p1 values less than (1991),
-	partition p3 values less than (2001)
+	partition p2 values less than (2001)
 	);`)
 
-	sql4 := `alter table t3 add partition (
+	sql4 := `alter table table3 add partition (
 		partition p3 values less than (1993)
 	);`
-	s.testErrorCode(c, sql4, mysql.ErrRangeNotIncreasing)
+	s.testErrorCode(c, sql4, tmysql.ErrRangeNotIncreasing)
 
-	sql5 := `alter table t3 add partition (
+	sql5 := `alter table table3 add partition (
 		partition p1 values less than (1993)
 	);`
-	s.testErrorCode(c, sql5, mysql.ErrSameNamePartition)
+	s.testErrorCode(c, sql5, tmysql.ErrSameNamePartition)
 
-	sql6 := `alter table t3 add partition (
+	sql6 := `alter table table3 add partition (
 		partition p1 values less than (1993),
 		partition p1 values less than (1995)
 	);`
-	s.testErrorCode(c, sql6, mysql.ErrSameNamePartition)
+	s.testErrorCode(c, sql6, tmysql.ErrSameNamePartition)
 
-	sql7 := `alter table t3 add partition (
+	sql7 := `alter table table3 add partition (
 		partition p4 values less than (1993),
-		partition p1 values less than (1995)，
-		partition p5 values less than maxvalue,
+		partition p1 values less than (1995),
+		partition p5 values less than maxvalue
 	);`
-	s.testErrorCode(c, sql7, mysql.ErrSameNamePartition)
+	s.testErrorCode(c, sql7, tmysql.ErrSameNamePartition)
+}
+
+func (s *testDBSuite) TestAlterTableDropPartition(c *C) {
```

Comment:
Could you add a test case like this:
1. Create a partitioned table
2. Insert some data (make sure  all partitions have some data)
3. Drop a partition
4. check the rows in that partition are gone and the rows in other partitions are still there

Reply:
Ok, I'll add the test.